File: src/config/libs/components/SampleEditor.ts
import { getComponent, initializePlugin } from '@/libs/ui';
import { ComponentProps } from './component';

export type Props = ComponentProps & {
};

export function SampleEditor({ bounds, track, ...props }: Props) {
    initializePlugin('SampleEditor', 'libzic_SampleEditorComponent.so');
    return getComponent('SampleEditor', bounds, [{ track }, props]);
}

File: config/libs/components/SampleEditor.lua
local ____lualib = require("lualib_bundle")
local __TS__ObjectRest = ____lualib.__TS__ObjectRest
local ____exports = {}
local ____ui = require("config.libs.ui")
local getComponent = ____ui.getComponent
function ____exports.SampleEditor(____bindingPattern0)
    local props
    local track
    local position
    position = ____bindingPattern0.position
    track = ____bindingPattern0.track
    props = __TS__ObjectRest(____bindingPattern0, {position = true, track = true})
    return getComponent("SampleEditor", position, {{track = track}, props})
end
return ____exports
